raiden/utils/cli.py

```diff
@@ -393,14 +401,25 @@ def apply_config_file(
     except TomlDecodeError as ex:
         raise ConfigurationError(f"Error loading config file: {ex}")
 
+    parsed_options = set()
     for config_name, config_value in config_file_values.items():
-        config_name_int = config_name.replace("-", "_")
-
-        if config_name_int not in paramname_to_param:
-            click.secho(
-                f"Unknown setting '{config_name}' found in config file - ignoring.", fg="yellow"
-            )
-            continue
+        try:
+            config_name_int = external_to_internal_map[config_name]
+        except KeyError:
+            raise KeyError(f"Unknown setting '{config_name}' found in config file")
```

Comment:
Using `RaidenUnrecoverableError` here would be better.
(Although I've just noticed that the config file parsing is done before the [try block catching that exception](https://github.com/raiden-network/raiden/blob/a7ef81db8a74460b1323460cd677ecc88887a46e/raiden/ui/cli.py#L642-L674), we should probably fix that).

Also using 
```
except KeyError as ex:
    raise RaidenUnrecoverableError(...) from ex
```

will give a slightly nicer traceback.

Reply:
Would you expect the `try:` to be moved to the beginning of the whole method?

Reply:
Your right, the KeyError definitely is not specific enough.
But why should it be an `RaidenUnrecoverableError` instead of an `ConfigurationError` like all others in that function?
Once it is put within the `try/catch`, the difference here is the stack-trace output for the unrecoverable. Is there a necessity for that with the config file parsing?

Reply:
Yes you're right, `ConfigurationError` is the proper exception class to use. 

> Once it is put within the try/catch, the difference here is the stack-trace output for the unrecoverable. Is there a necessity for that with the config file parsing?

No but that long except block takes care to use a proper exit code and also the somewhat nicely format the exception. 
Tracebacks should usually not be shown to end users for "known" errors.

Reply:
Ok perfect, then we're on the same page :)